Oracle-Validation.md

```diff
@@ -0,0 +1,100 @@
+requirements, which should be swayed from only if protocol has higher built-in privacy mechanisms.
+
+## Requirements
+
+A node:
+* MUST persist its keyring across restart/shutdown.
+* SHOULD fetch any `oracle_public_key` advertised by a "good reputation" oracle.
+
+## Rationale
+
+A DLC client should desynchronize any oracle pubkey fetching from DLC contract setting, otherwise
+the fetched entity may deduce an usage of oracle upcoming events. By fetching any `oracle_public_key`
+ available, a DLC client is hindering better potential oracle usage among a wider anonymous set. 
+"Good-reputation" is enforced by the client oracle selection policy and its definition is beyond this
+current specification.
+
+# `oracle_announcement` Authentication
+
+[`oracle_announcement`](https://github.com/discreetlogcontracts/dlcspecs/blob/master/Oracle.md#oracle-announcements)  
+authentication happens at `offer_dlc` reception, thus verifying binding between an event and an
+oracle.
+
+## Requirements
+
+A node:
+- SHOULD NOT fetch `oracle_public_key` if it's not already a member of client oracle keyring
+- if `oracle_public_key` has not been accepted by client oracle selection policy:
+    - MUST reject the `offer_dlc`
+- if `oracle_event` signature is not valid against the `oracle_public_key`:
+    - MUST reject the `offer_dlc`
+
+## Rationale
+
+Fetching `oracle_public_key` at `offer_dlc` is a privacy leak vector for the message receiver.
+Assuming oracle keys fetching is done through some privacy-preserving channel, if the DLC 
+counterparty can trigger communications through this channel it might hinder DLC client anonymity
+toward the oracle.
+
+Rejecting announcement for which the `oracle_public_key` hasn't been accepted by client oracle
+selection policy and announcement hasn't been authenticated prevent usage of lazy or malicious
+oracles. Authenticating event also prevent entering in a DLC position for which the event won't
+be signed, and thus losing timevalue of collaterals.
+
+# `oracle_attestation` Authentication
+
+## Requirements
+
+A node:
+- if an `oracle_attestation` has not been requested by the DLC client:
+   - MUST ignore the `oracle_attestation`, even if message is valid
+- if an announcement for this `event_id` hasn't been previously accepted OR announcement's `oracle_public_key` and attestation's `oracle_public_key` are not equal:
+   - MUST ignore the `oracle_attestation`
+- if the range of `signatures` is not valid against the nonces committed in the accepted announcement
+   - MUST ignore the `oracle_attestation`
+
+## Rationale
+
+Rejecting unrequested `oracle_attestation` prevents attestations injection by an attacker to discover
+other oracles actually consumed by this DLC client. Accepting unrequested _valid_ `oracle_attestation`
+and broadcasting a CET transaction on the p2p network are observable behaviors revealing DLC presenve
+for the corresponding event and oracle.
```

Comment:
I don't really follow this part. Do you mean an unrequested `oracle_attestation` but for which you actually have a DLC? In that case I'm not sure how we could propagate attestation in a P2P setting.

Reply:
It's all depend of we propagate messages in this p2p settings. 

If it's something similar to tx-relay on the base layer, you do have an inventory-getdata-tx messages sequence. If the inventory is already known by receiver you won't emit a getdata to save on bandwidth/CPU.

If it's something like Lightning gossips, without any announcement, effectively this requirement would be a blocker to process attestations. 

So yeah I agree, I think that's a small mitigation against DoS assuming our p2p network fits the first model. Likely to early as a reasoning :sweat_smile:

Reply:
Thanks for the clarification :)